### ISSUE TITLE
Make core libraries platform-agnostic

### ISSUE THREAD

**Issue by kumavis** [MetaMask/metamask-plugin#1291]
in order for metamask to work on different platforms / in different environments ( extension, metamascara, electron... ), we need to move all the platform dependent parts out of the core libraries

main offender: `metamaskController`

some notes:

```
  ui
    connect to background
    notification
  proxy
    connect to background
  background
    notification
    metamaskController

  notification
    window mgmt
      extension.windows

  metamaskController
    window mgmt
      extension.tabs.create({ url })
    set rpc provider
      extension.runtime.reload()
```

**Comment by frankiebee** [comment]
related to #775

**Comment by kumavis** [comment]
closed as duplicate of #775
